qp_shotgun/shogun/shogun.py

```diff
@@ -0,0 +1,174 @@
+        parameter = func_params[param]
+        value = parameters[param]
+        params[parameter] = value
+
+    return params
+
+
+def generate_shogun_align_commands(input_fp, temp_dir, parameters):
+    cmds = []
+    cmds.append(
+        'shogun align --aligner {aligner} --threads {threads} '
+        '--database {database} --input {input} --output {output}'.format(
+            aligner=parameters['aligner'],
+            threads=parameters['threads'],
+            database=parameters['database'],
+            input=input_fp,
+            output=temp_dir))
+
+    return cmds
+
+
+def generate_shogun_assign_taxonomy_commands(temp_dir, parameters):
+    cmds = []
+    aln2ext = {'utree': 'tsv', 'burst': 'b6', 'bowtie2': 'sam'}
+    ext = aln2ext[parameters['aligner']]
+    output_fp = join(temp_dir, 'profile.tsv')
+    cmds.append(
+        'shogun assign_taxonomy '
+        '--aligner {aligner} '
+        '--database {database} '
+        '--input {input} --output {output}'.format(
+            aligner=parameters['aligner'],
+            database=parameters['database'],
+            input=join(temp_dir, 'alignment.%s.%s' % (parameters['aligner'],
+                                                      ext)),
+            output=output_fp))
+
+    return cmds, output_fp
+
+
+def generate_shogun_functional_commands(profile_dir,
+                                        temp_dir, parameters):
+    cmds = []
+    cmds.append(
+        'shogun functional '
+        '--database {database} '
+        '--input {input} '
+        '--output {output} '
+        '--level species'.format(
+            database=parameters['database'],
+            input=profile_dir,
+            output=temp_dir))
+
+    return cmds
+
+
+def shogun(qclient, job_id, parameters, out_dir):
+    """Run Shogun with the given parameters
+
+    Parameters
+    ----------
+    qclient : tgp.qiita_client.QiitaClient
+        The Qiita server client
+    job_id : str
+        The job id
+    parameters : dict
+        The parameter values to run split libraries
+    out_dir : str
+        The path to the job's output directory
+
+    Returns
+    -------
+    bool, list, str
+        The results of the job
+    """
+    # Step 1 get the rest of the information need to run Atropos
+    qclient.update_job_step(job_id, "Step 1 of 6: Collecting information")
+    artifact_id = parameters['input']
+    del parameters['input']
+
+    # Get the artifact filepath information
+    artifact_info = qclient.get("/qiita_db/artifacts/%s/" % artifact_id)
+    fps = artifact_info['files']
+
+    # Get the artifact metadata
+    prep_info = qclient.get('/qiita_db/prep_template/%s/'
+                            % artifact_info['prep_information'][0])
+    qiime_map = prep_info['qiime-map']
+
+    # Step 2 converting to fna
+    qclient.update_job_step(
+        job_id, "Step 2 of 6: Converting to FNA for Shogun")
+    temp_path = os.environ['QC_SHOGUN_TEMP_DP']
```

Comment:
Do we really need this new folder? Or can we simply use do something like:
`temp_path = join(out_dir, 'tmp')`
Note that the difference is that if we use QC_SHOGUN_TEMP_DP the folder is gonna be always created in that path, which is in home/where-the-software-was-installed but if we do it like in my suggestion the path is gonna be created in the mount/job-folder.

Reply:
The idea was that we can change the base temp path by changing the environment variable without having to reload the whole plugin. However, it makes sense to simply make the temp directory in the output